rubin_jupyter_utils/helpers/functions.py

```diff
@@ -327,44 +322,87 @@ def add_user_to_groups(uname, grpstr, groups=["lsst_lcl", "jovyan"]):
     return g_str
 
 
-def get_pull_secret(cfg):
-    if not (cfg.lab_repo_password and cfg.lab_repo_username):
+def load_k8s_config(log=None):
+    try:
+        load_incluster_config()
+    except ConfigException:
+        if not log:
+            log = make_logger()
+        log.warning("In-cluster config failed! Falling back to kube config.")
+        try:
+            load_kube_config()
+        except ValueError as exc:
+            log.error("Failed to load k8s config: {}".format(exc))
+            raise
+
+
+def build_pull_secret(username, password, pull_secret_name="pull-secret"):
+    if not (username and password):
         return None  # These do not exist unless we have both auth parts
-    basic_auth = '{}:{}'.format(cfg.lab_repo_username,
-                                cfg.lab_repo_password).encode('utf-8')
+    basic_auth = "{}:{}".format(username, password).encode("utf-8")
     authdata = {
         "auths": {
             cfg.lab_repo_host: {
-                "username": cfg.lab_repo_username,
-                "password": cfg.lab_repo_password,
-                "auth": base64.b64encode(basic_auth).decode('utf-8')
+                "username": username,
+                "password": password,
+                "auth": base64.b64encode(basic_auth).decode("utf-8"),
             }
         }
     }
-    b64authdata = base64.b64encode(json.dumps(
-        authdata).encode('utf-8')).decode('utf-8')
+    b64authdata = base64.b64encode(
+        json.dumps(authdata).encode("utf-8")
+    ).decode("utf-8")
     pull_secret = client.V1Secret()
-    pull_secret.metadata = client.V1ObjectMeta(name='pull-secret')
+    pull_secret.metadata = client.V1ObjectMeta(name=pull_secret_name)
     pull_secret.type = "kubernetes.io/dockerconfigjson"
     pull_secret.data = {".dockerconfigjson": b64authdata}
     return pull_secret
 
 
-def get_pull_secret_reflist(pull_secret):
-    if not pull_secret:
+def get_pull_secret(pull_secret_name="pull-secret", api=None, log=None):
+    if not pull_secret_name:
+        return
+    ns = get_execution_namespace()
+    if not api:
+        load_k8s_config()
+        api = CoreV1Api()
+    try:
+        secret = api.read_namespaced_secret(pull_secret_name, ns)
+    except ApiException as e:
+        if not log:
```

Comment:
why bother with passing all this log stuff?  why not just make one for the module?

Reply:
'Cause it's a function rather than a module, and I would rather the logger identify itself as the logger from the thing calling the function, but I specifically want to log the 409 case, which happens in the function.